R/available_forecasts.R

```diff
@@ -34,14 +34,15 @@
 #'   data.table::setDTthreads(2) # restricts number of cores used on CRAN
 #' }
 #'
-#' get_forecast_counts(example_quantile,
+#' get_forecast_counts(
+#'   as_forecast(example_quantile),
 #'   by = c("model", "target_type")
 #' )
 get_forecast_counts <- function(data,
                                 by = NULL,
                                 collapse = c("quantile_level", "sample_id")) {
-
-  data <- as_forecast(data)
+  data <- copy(data)
```

Comment:
some places use ensure_data.table and some places use this?

Reply:
So my internal logic was that 
- functions that require a forecast object use `copy()` and
- functions that can accept any input use `ensure_data.table()`

But as per the comment above we can probably replace this by just having `validate_forecast()` produce a copy.